app/views/runs/_timeline.slim

```diff
@@ -4,12 +4,13 @@
 - else
   - if run.video&.supports_embedding?
     div style="height: 1em": span.video-progress-line.timeline-animation id="video-progress-line-#{run.id36}"
-  .timeline-background.timeline-animation data={run_id: run.id36}
+  - collapsed_segments = run.collapsed_segments(timing)
+  .timeline-background.timeline-animation data={run_id: run.id36, segment_names: collapsed_segments.map(&:display_name), total_segments: collapsed_segments.length}
     .timeline.shadow style="width: #{run.duration(timing) / scale_to * 100.0}%"
-      - run.collapsed_segments(timing).each.with_index do |segment, index|
+      - collapsed_segments.each.with_index do |segment, index|
         .pure-u.split(
           id="#{run.id36}-split-#{index}"
-          data={start_ms: segment.start(timing).to_ms, run_id: run.id36, segment_number: index+1}
+          data={start_ms: segment.start(timing).to_ms, run_id: run.id36, segment_name: segment.display_name, segment_number: index+1, total_segments: collapsed_segments.length}
```

Comment:
Seems like we're duplicating information here:
- The array of segment names is stored on `.timeline`, but it can be fetched with 
    ```javascript
    Array.from(timeline.getElementsByClassName('split')).map(el => el.dataset.segment_name)
    ```
- The number of segments is being stored on both `.timeline` and every `.split`, but it can be fetched with
    ```javascript
    timeline.getElementsByClassName('split').length
    ```

Reply:
Sounds good. I hate redoing work over and over, but storing all the names and counts multiple times is also bad.

Reply:
Yeah I understand the struggle. In general we prefer extra work over extra storage on Splits.io, if it's not clear-cut which is better.